### ISSUE TITLE
IntelliSense suggestions by passing a typescript object

### ISSUE THREAD

**Issue by iosifnicolae2** [suren-atoyan/monaco-react#540]
**Is your feature request related to a problem? Please describe.**
Yes, we would like to have IntelliSense suggestions by passing the application context using an object.

Codemirror has a method called `scopeCompletionSource` which parse the structure of an object and gives the editor informaiton about which types are available (https://github.com/codemirror/lang-javascript#user-content-scopecompletionsource).

**Describe the solution you'd like**
We would like to pass our context and the IntelliSense should suggest the right type.
Example:
```typescript
import React from "react";

const EditorComponent = (
  <Editor
    language="javascript"
    scope={{React}}
  />
);

``` 

**Describe alternatives you've considered**
Codemirror with `scopeCompletionSource` but unfortunetly, it doesn't work very well.

Thank you!

**Comment by iosifnicolae2** [comment]
Linked solution (unfortunetly is not working 100%): https://gist.github.com/mwrouse/05d8c11cd3872c19c684bd1904a2202e

**Comment by suren-atoyan** [comment]
> Linked solution (unfortunetly is not working 100%): https://gist.github.com/mwrouse/05d8c11cd3872c19c684bd1904a2202e

yes, this is a good start. I implemented my own version based on this, I'll try to find and share it

**Comment by suren-atoyan** [comment]
> > Linked solution (unfortunetly is not working 100%): https://gist.github.com/mwrouse/05d8c11cd3872c19c684bd1904a2202e
> 
> yes, this is a good start. I implemented my own version based on this, I'll try to find and share it

but I used that with python code, with TypeScript might be a better solution
